File: src/Rhino.java
public class Rhino extends Pachyderm {

    Rhino(){
        super(true, "vegetation", "savannah", true, "Ronald", 4, 1);
    }

    /**
     *  Constructor with only Animal parameters
     * @param awake:boolean - Is the animal awake?
     * @param diet:String - What does the animal eat?
     * @param habitat:String - Where does the animal live?
     * @param hungry:boolean - Is the animal hungry?
     * @param name:String - The name of the animal
     * @param numLegs:int - How many legs does the animal have?
     */
    Rhino(boolean awake, String diet, String habitat, boolean hungry, String name, int numLegs){
        super(awake, diet, habitat, hungry, name, numLegs);
    }

    /**
     *  Constructor with Animal and Pachyderm parameters
     * @param awake:boolean - Is the animal awake?
     * @param diet:String - What does the animal eat?
     * @param habitat:String - Where does the animal live?
     * @param hungry:boolean - Is the animal hungry?
     * @param name:String - The name of the animal
     * @param numLegs:int - How many legs does the animal have?
     *
     * @param numTusks:int - How many tusks does the animal possess?
     */
    Rhino(boolean awake, String diet, String habitat, boolean hungry, String name, int numLegs, int numTusks){
        super(awake, diet, habitat, hungry, name, numLegs, numTusks);
    }

    /**
     * override roam for rhinos
     */
    @Override
    public void roam() {
        double rand = Math.random();
        if (rand < 0.5){
            System.out.println(name + " the Rhino: Charges! How Scary.");
        } else {
            System.out.println(name + " the Rhino: Runs around his paddock.");
        }
    }

    /**
     * override make noise for rhino noises
     */
    @Override
    public void makeNoise() { System.out.println(name + " the Rhino: *screams and trumpets*");}
}

File: Project2/Rhino.py
import random
from Pachyderm import Pachyderm

class Rhino(Pachyderm):

  '''
    Generic initializer for Rhino
    @param name:String, default:'Raymond'
    @param awake:Boolean, default:False
    @param hungry:Boolean, deault:True
    @param num_legs:Int, default:4
  '''
  def __init__(self, name='Raymond', awake=False, hungry=True, num_legs=4):
    super(Rhino, self).__init__(name, awake, hungry, num_legs)

  '''
    Rhino has a 1/3 chance to scream, trumpet, or scream and trumpet
  '''
  def make_noise(self):
    animal_type = type(self).__name__
    random_number = random.random()
    if random_number < 0.33:
      print('{} the {} screams.'.format(self.name, animal_type))
    elif random_number <  0.67:
      print('{} the {} trumpets.'.format(self.name, animal_type))
    else:
      print('{} the {} screams and trumpets.'.format(self.name, animal_type))
      

